proto/cluster.proto

```diff
@@ -161,6 +161,9 @@ message Cluster {
   
   // Output. The service endpoint of the cluster
   map<string, string> service_endpoint = 11;
+
+  // Output. Various metadata about the cluster.
+  map<string, string> metadata = 12;
```

Comment:
em.. We actually have a metadata field in our downstream and it's reserved for our company specific setting..
Trying to see your case and if we can make it generic

![image](https://user-images.githubusercontent.com/4739316/181125259-c6295144-e48d-4c8e-979b-9c3594382f7c.png)

Reply:
yes, metadata is a map, but it includes certain attributes in addition to nested maps inside of it.

It is not usual to have a metadata in status. we need to really have a very strong case for that. if we simply need a map to add information in status, we can simply add a regular map, not the metadata object. 

I hope that clarifies more.

Reply:
I see. I'll add metadata to K8s annotations instead.

Reply:
do you mean you will use K8s annotation instead of metadata?

Reply:
nit: here the `metadata` terms is a little bit confusing, do we need rename it to be differentiated from kubernetes native metadata?

Reply:
> do you mean you will use K8s annotation instead of metadata?

@akanso yes

> nit: here the `metadata` terms is a little bit confusing, do we need rename it to be differentiated from kubernetes native metadata?

@scarlet25151 good point, any suggestions?

Reply:
@davidxia for now, I have no better idea, what about other folks thinks?

Reply:
how about just `data` :)

Reply:
that works. I'll close this PR and start a new one soon then. But feel free to discuss more here